cmd/ipfs/daemon.go

```diff
@@ -227,11 +226,7 @@ func daemonFunc(req cmds.Request, res cmds.Response) {
 
 	if initialize {
 
-		// now, FileExists is our best method of detecting whether ipfs is
-		// configured. Consider moving this into a config helper method
-		// `IsInitialized` where the quality of the signal can be improved over
-		// time, and many call-sites can benefit.
-		if !util.FileExists(req.InvocContext().ConfigRoot) {
+		if !fsrepo.IsInitialized(req.InvocContext().ConfigRoot) {
```

Comment:
Instead of calling twice the ```req.InvocContex().ConfigRoot``` why not make under initialize something like 
```go
if initialize {
   cfg := req.InvocContext()
   if !fsrepo.IsInitialized(cfg) {
       err := initWithDefaults(os.Stdou, cfg)
// code
```